lib/redis/distributed_store.rb

```diff
@@ -6,12 +6,12 @@ class DistributedStore < Distributed
     attr_reader :ring
 
     def initialize(addresses, options = { })
-      nodes = addresses.map do |address|
-        ::Redis::Store.new _merge_options(address, options)
-      end
-
       _extend_namespace options
-      @ring = Redis::HashRing.new nodes
+      @ring = options[:ring] || Redis::HashRing.new([], options[:replicas] || Redis::HashRing::POINTS_PER_SERVER)
```

Comment:
If `HashRing` took keyword arguments and defaulted to `nodes=[]`, you could instantiate one like so:

```ruby
Redis::HashRing.new(replicas: options[:replicas], )
```

Reply:
Right, but that still doesn't work if I want to pass in my own custom HashRing object (without any nodes populated yet).

say I have a class called RandomlyPutNodesInRingRing

my_ring = RandomlyPutNodesInRingRing.new
redis_store = Redis::Store::Factory.create(nodes_from_my_config, ring: my_ring)

that will pass the ring in, with no nodes in it, and then will create and add the nodes from my config. the Redis::HashRing constructor will never get called

Reply:
Doh, I think you understood that. I totally didn't read your comment right. And yes, that would be nice, but it doesn't. So I have to pass an empty array

Reply:
no you don't. https://github.com/redis/redis-rb/blob/master/lib/redis/hash_ring.rb#L13

Reply:
I'm not sure what you're asking here. The current state of Redis::HashRing.new is two positional parameters. If I want to pass something to the second parameter (I do!), I have to supply the first parameter, so I'm passing exactly what it defaults to. Yes, it would be nice if it took keyword params instead of positional parameters, so that I could just pass the one arg I care about, but it doesn't. Are you wanting me to try and get a keyword arg pull request accepted to redis-rb?

Reply:
i guess that would be kind-of a breaking change.